### PR TITLE
Add invalid argument exception

### PR DESCRIPTION
**Pull request by gmponos**
Hello there,

Let's say I have the following class:

``` php
class MyClientWrapper {
    private $httpClient;

    public function callSomething(){
          $response = $this->httpClient->get('/');
          return GuzzleHttp\json_decode($response->getBody());
    }
}
```

when I call the function above I have to do something like this:

``` php
try{
   $mywrapper->callSomething();
} catch (\InvalidArgumentException | GuzzleException $e){
    ....
}
```

This has the defect that I if another part of my code throws an `InvalidArgumentException` inside this class I can not separate it from the the GuzzleHttp `InvalidArgumentException`.

### CODE REVIEW

File: src/Exception/InvalidArgumentException.php [hunk 1]
```diff
@@ -0,0 +1,5 @@
+<?php
+
+namespace GuzzleHttp\Exception;
+
+class InvalidArgumentException extends \InvalidArgumentException implements GuzzleException {}
```
Comment (sagikazarmark) [inline src/Exception/InvalidArgumentException.php]: I think this class should be final for now, we can always open it up later.

Also, please add a trailing new line to the end of the file.

Comment (gmponos) [reply src/Exception/InvalidArgumentException.php]: Done.. I added the new line but I am not sure if it is visible through the PR changes tab

**Comment by FrontEndCoffee** [comment]
Looks good 👍

**Comment by sagikazarmark** [comment]
Looks good as always @gmponos ! Thanks!

**Comment by wirwolf** [comment]
Hello @gmponos, i create a fork of this repo and release this marge request into the new version(6.4.0).

Namespace compatibility is 100%, and you can use my changes if you patch composer.json in your projects
* replace `"guzzlehttp/guzzle": "^6.3"` to `"someblackmagic/guzzle": "^6.4"` 
* add `"guzzlehttp/guzzle": "6.3.*"` into replace block

Fork link: https://github.com/SomeBlackMagic/guzzle/

**Pull request merged**